### PR TITLE
(DOCSP-25395): Update App Services docs for Client Reset with Recovery

### PR DESCRIPTION
**Pull request by dacharyc**
### Jira

- https://jira.mongodb.org/browse/DOCSP-25395

### Staged Changes (Requires MongoDB Corp SSO)

- [Atlas Device Sync Configuration Files](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/DOCSP-25395/manage-apps/configure/config/sync/)
- [Enable Atlas Device Sync](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/DOCSP-25395/sync/configure/enable-sync/)
- [Pause or Terminate Sync](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/DOCSP-25395/sync/configure/pause-or-terminate-sync/)
- [Update Your Schema](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/DOCSP-25395/sync/data-model/update-schema/)
- [Client Resets](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/DOCSP-25395/sync/error-handling/client-resets/)
- [Sync Errors](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/DOCSP-25395/sync/error-handling/errors/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-app-services/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sync/configure/enable-sync.txt
Comment (mongomoe) [inline source/sync/configure/enable-sync.txt:150]: Might want to link to [link a data source](https://www.mongodb.com/docs/atlas/app-services/mongodb/#link-a-data-source) here

File: source/sync/configure/enable-sync.txt
Comment (mongomoe) [inline source/sync/configure/enable-sync.txt:283]:
```suggestion
            performance and improve the data recovery process on the client. These 
```

Suggestion/Not Required for LGTM:
The phrase "client data recovery process" sounds a bit less natural than "data recovery process on the client" even if they are similar.

File: source/sync/configure/enable-sync.txt
Comment (mongomoe) [inline source/sync/configure/enable-sync.txt:886]:
```suggestion
            your App, this should be a service whose ``name``
```

File: source/sync/error-handling/client-resets.txt
Comment (mongomoe) [inline source/sync/error-handling/client-resets.txt:94]:
```suggestion
#. Performs a diff to figure out the steps required to
```

File: source/sync/error-handling/client-resets.txt
Comment (mongomoe) [inline source/sync/error-handling/client-resets.txt:102]:
```suggestion
   any inserts, updates, and deletes from the backend that hasn't yet 
```

File: source/sync/error-handling/client-resets.txt
Comment (mongomoe) [inline source/sync/error-handling/client-resets.txt:150]:
```suggestion
locally that did not sync before the client reset. These rules 
```
needed fix

File: source/sync/error-handling/client-resets.txt
Comment (mongomoe) [inline source/sync/error-handling/client-resets.txt:152]: Should this paragraph specify that deletions take priority like the bullet points mention.

File: source/sync/error-handling/client-resets.txt
Comment (mongomoe) [inline source/sync/error-handling/client-resets.txt:208]: I don't understand that first sentence, why would a manual client reset handler tell the user to update the app? Seems like the developer should create code in this handler to update the objects in response to the breaking change. Am I misunderstanding something?

File: source/sync/configure/enable-sync.txt
Comment (mongomoe) [inline source/sync/configure/enable-sync.txt:299]:
```suggestion
            If you have disabled recovery mode and want to re-enable it, set ``is_recovery_mode_disabled`` to 
```

File: source/sync/configure/enable-sync.txt
Comment (dacharyc) [reply source/sync/configure/enable-sync.txt:886]: I want to be very explicit about these values and don't mind being repetitive in this instance, so I'm going to leave this one in.

File: source/sync/data-model/update-schema.txt
Comment (dacharyc) [reply source/sync/data-model/update-schema.txt:275]: This page is written so devs can select the type of change they want to make and see how to deal with it. I don't expect most people will read the page from top to bottom; I expect they'll see the table, go "Oh, I want to change this property from optional to required" (for example) and then jump to that section and read it. All of the information that they need should be contained in the section that's relevant to them, because we can't assume they'll read any other sections of the page.

File: source/sync/error-handling/client-resets.txt
Comment (dacharyc) [reply source/sync/error-handling/client-resets.txt:102]: "Haven't" in this instance refers to "inserts" "updates" and "deletes", so the plural is the correct form to use, grammatically-speaking.

File: source/sync/error-handling/client-resets.txt
Comment (dacharyc) [reply source/sync/error-handling/client-resets.txt:208]: We are assuming that you have made a breaking change intentionally, and you have released a new version of the app to handle the new models. Per engineering, the advice for the developer in this instance is to tell users to update to the new version of the app, since there is no meaningful way to recover from this type of error.

**Comment by sudssm** [comment]
image

wording is incorrect here - should say "Select a Cluster to Sync" to match the screenshot

File: source/sync/error-handling/client-resets.txt
Comment (sudssm) [inline source/sync/error-handling/client-resets.txt:133]: Client Recovery (or Automatic Client Recovery)

File: source/sync/error-handling/client-resets.txt
Comment (sudssm) [reply source/sync/error-handling/client-resets.txt:105]: I see, you're specifically talking about `recoverOrDiscard` here. Might be more clear that this is a different mode if `recoverOrDiscard` was introduced at the top of the paragraph?

**Review by sudssm — COMMENTED** [review]
This should also get a once-over from someone on the Realm Core team - @jbreams or someone on his team

File: source/sync/data-model/update-schema.txt
Comment (mongomoe) [reply source/sync/data-model/update-schema.txt:275]: Good point

**Review by mongomoe — APPROVED** [review]
LGTM

File: source/sync/configure/enable-sync.txt
Comment (dacharyc) [reply source/sync/configure/enable-sync.txt:45]: Yeah, we call out on line 42 above "unless you are using Development Mode" with a link to the dev mode docs for those who may not be familiar with it. Your note here makes me wonder if we should explicitly state here that you can skip defining a schema if you have Dev Mode enabled.

File: source/sync/configure/enable-sync.txt
Comment (dacharyc) [reply source/sync/configure/enable-sync.txt:555]: Ah, yeah, that's probably an artifact of different people adding these at different times and not looking at them together. I'll make these consistent. Good catch!

File: source/sync/configure/enable-sync.txt
Comment (dacharyc) [reply source/sync/configure/enable-sync.txt:1049]: Nice catch - thanks!

File: source/sync/data-model/update-schema.txt
Comment (dacharyc) [reply source/sync/data-model/update-schema.txt:658]: I was under the impression that the automatic client reset handler couldn't handle breaking changes, so you still need to define a manual handler for cases where you make a breaking change. That's why I used this language here and anywhere we talk about breaking changes. Is that not the case?

File: source/sync/error-handling/client-resets.txt
Comment (dacharyc) [reply source/sync/error-handling/client-resets.txt:105]: We wanted to discuss the various cases for handling schema changes, since that is probably more meaningful to devs reading the docs than starting out the discussion with a feature name that devs might not have encountered - i.e. recoverOrDiscard.

File: source/sync/data-model/update-schema.txt
Comment (sudssm) [reply source/sync/data-model/update-schema.txt:658]: yep, you're correct on that one. I guess my objection was to the phrase "manual client reset", which doesn't have too much meaning imo - I think you want to say "your client application will execute the manual client reset handler" (which would then have to handle the breaking change...

Comment (mongodben) [reply source/sync/data-model/update-schema.txt:658]: i think it'd make sense to keep this sentence as is. then provide another short sentence in line with what Sudarshan is saying, with a brief high lvl overview of how this is implemented. perhaps something like:

```rst
To enable your client app to execute a manual client reset, add a manual client reset handler using a Realm SDK.
```

^ this sentence uses the word 'client' a lot, so should probably be refactored a bit. I leave that in your very able hands @dacharyc  😄

Comment (dacharyc) [reply source/sync/data-model/update-schema.txt:658]: Alrighty, went with this additional line:

> You can define an error handler in your client application code to execute some custom logic in this scenario.

Thanks for the feedback, both!

**Pull request merged**
